aws/image/misc-settings.sh

```diff
@@ -20,24 +20,95 @@ if [ "$CLOUD_PROVIDER" == "vsphere" ]
 then
 # Add vsphere specific commands to execute on remote
     case $PODVM_DISTRO in
-	centos)
-	    #fallthrough
-	    ;&
-	rhel)
-	    (! dnf list --installed | grep open-vm-tools >/dev/null 2>&1) && \
-		(! dnf -y install open-vm-tools) && \
-		     echo "$PODVM_DISTRO: Error installing package required for cloud provider: $CLOUD_PROVIDER" 1>&2 && exit 1
-	    ;;
-	ubuntu)
-	    (! dpkg -l | grep open-vm-tools > /dev/null 2>&1) && apt-get update && \
-		(! apt-get -y install open-vm-tools > /dev/null 2>&1) && \
-		     echo "$PODVM_DISTRO: Error installing package required for cloud provider: $CLOUD_PROVIDER" 1>&2  && exit 1
-	    ;;
-	*)
-	    ;;
+    centos)
+        #fallthrough
+        ;&
+    rhel)
+        (! dnf list --installed | grep open-vm-tools > /dev/null 2>&1) && \
+        (! dnf -y install open-vm-tools) && \
+             echo "$PODVM_DISTRO: Error installing package required for cloud provider: $CLOUD_PROVIDER" 1>&2 && exit 1
+        ;;
+    ubuntu)
+        (! dpkg -l | grep open-vm-tools > /dev/null 2>&1) && apt-get update && \
+        (! apt-get -y install open-vm-tools > /dev/null 2>&1) && \
+             echo "$PODVM_DISTRO: Error installing package required for cloud provider: $CLOUD_PROVIDER" 1>&2  && exit 1
+        ;;
+    *)
+        ;;
     esac
-# else if...
-#
+fi
+
+# Setup oneshot systemd service for AWS and Azure to enable NAT rules
+if [ "$CLOUD_PROVIDER" == "azure" ] || [ "$CLOUD_PROVIDER" == "aws" ]
+then
+# Create a oneshot systemd service to execute setup-nat-for-imds.sh
+# during first boot
+cat <<END > /etc/systemd/system/setup-nat-for-imds.service
+[Unit]
+Description=Setup NAT for IMDS
+After=kata-agent.service
+Wants=kata-agent.service
+
+[Service]
+Type=oneshot
+ExecStart=/usr/local/bin/setup-nat-for-imds.sh
+
+[Install]
+WantedBy=multi-user.target
+END
+
+# Add a link to the service in multi-user.target.wants
+ln -s /etc/systemd/system/setup-nat-for-imds.service /etc/systemd/system/multi-user.target.wants/setup-nat-for-imds.service
+
+# Create a script to setup NAT for IMDS
+cat <<END > /usr/local/bin/setup-nat-for-imds.sh
+#!/bin/bash
+# This script sets up NAT for IMDS on Azure/AWS
+# This is required for IMDS to work on Azure/AWS
+# This script is executed as a oneshot systemd service
+# during first boot
+
+set -euo pipefail
+
+IMDS_IP="169.254.169.254"
+DUMMY_IP="169.254.99.99"
+
+# trap errors
+trap 'echo "Error: \$0:\$LINENO stopped"; exit 1' ERR INT
+
+# Function to setup veth pair
+function setup_proxy_arp() {
+  local pod_ip=${pod_ip:-}
```

Comment:
I think this is harmless, but it can simply be like this, since `pod_ip` is overwritten by the next line?
```
local pod_ip
```